src/components/url-form.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
```

Comment:
lets abstract all types and zod schemas into seperate files to make things cleaner. lib/schema.ts lib/types.ts makes sense for now